test/drivers/test_revd2.cc

```diff
@@ -14,69 +14,94 @@ class TestREVD2 : public ::testing::Test
 
     virtual void TearDown() {};
 
-    /// General test for RSVD:
-    /// Computes the decomposition factors, then checks A-U\Sigma\transpose{V}.
-    template <typename T, typename RNG>
-    static void test_REVD2_general(int64_t m, int64_t k, int64_t k_start, std::tuple<int, T, bool> mat_type, RandBLAS::base::RNGState<RNG> state, int rank_expectation, T err_expectation) {
+    template <typename T>
+    struct REVD2TestData {
+        std::vector<T> A_buf;
+        std::vector<T> A;
+        std::vector<T> A_cpy;
+        std::vector<T> V;
+        std::vector<T> eigvals;
+        std::vector<T> E;
+        std::vector<T> Buf;
+
+        REVD2TestData(int64_t m, int64_t k) :
+        A_buf(m * m, 0.0), 
+        A(m * m, 0.0), 
+        A_cpy(m * m, 0.0),  
+        V(m * k, 0.0), 
+        eigvals(k, 0.0), 
+        E(k * k, 0.0), 
+        Buf(m * k, 0.0)
+        {}
+    };
 
-        printf("|==================================TEST REVD2 GENERAL BEGIN==================================|\n");
-
-        // For running QB
-        std::vector<T> A_buf(m * m, 0.0);
-        RandLAPACK::util::gen_mat_type(m, m, A_buf, k, state, mat_type);
+    template <typename T, typename RNG>
+    struct algorithm_objects {
+        RandLAPACK::PLUL<T> Stab;
+        RandLAPACK::RS<T, RNG> RS;
+        RandLAPACK::HQRQ<T> Orth_RF;
+        RandLAPACK::RF<T> RF;
+        RandLAPACK::REVD2<T, RNG> REVD2;
+
+        algorithm_objects(bool verbosity, 
+                            bool cond_check, 
+                            int64_t p, 
+                            int64_t passes_per_iteration, 
+                            RandBLAS::base::RNGState<RNG> state) :
+                            Stab(cond_check, verbosity),
+                            RS(Stab, state, p, passes_per_iteration, verbosity, cond_check),
+                            Orth_RF(cond_check, verbosity),
+                            RF(RS, Orth_RF, verbosity, cond_check),
+                            REVD2(RF, state, verbosity)
+                            {}
```

Comment:
I'll ask that you format this differently for improved readability
```
algorithm_objects(
    bool verbosity, 
    bool cond_check, 
    int64_t p, 
    int64_t passes_per_iteration, 
    RandBLAS::base::RNGState<RNG> state
) : 
    Stab(cond_check, verbosity),
    RS(Stab, state, p, passes_per_iteration, verbosity, cond_check),
    Orth_RF(cond_check, verbosity),
    RF(RS, Orth_RF, verbosity, cond_check),
    REVD2(RF, state, verbosity)
    {}
```

Reply:
Fixed.